src/pages/LoginHandler/index.js

```diff
@@ -9,138 +9,102 @@ import {
   Title,
 } from '@patternfly/react-core';
 import AuthLayout from '@/components/AuthLayout';
-import { routerRedux } from 'dva/router';
 import { connect } from 'dva';
 import styles from './index.less';
+import { validateEmail } from '@/utils/validator';
 
-@connect(auth => ({
-  auth: auth.auth,
-}))
-class LoginHandler extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      username: '',
-      password: '',
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-    };
-  }
+const mapStateToProps = state => {
+  const { auth } = state;
+  return { auth };
+};
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
-  };
+const LoginHandler = props => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [errors, setErrors] = useState({
+    email: '',
+  });
+  const [btnDisabled, setBtnDisabled] = useState(true);
 
-  setLoggedIn = payload => {
-    const { dispatch } = this.props;
-    dispatch({
-      type: 'auth/loadUser',
-      payload,
+  const handleUsernameChange = val => {
+    setUsername(val);
+    const validEmail = validateEmail(val);
```

Comment:
We should consider disabling the submit button to prevent bad requests going to the API anytime the email is invalid. We perform a similar check for form validation for the sign up flow that we can adopt here.

Reply:
Ah, that's right! Thanks for pointing that out. Just made the changes @gurbirkalsi